### PR TITLE
Add Navigation to the app, as well as some basics screens, update gradle build version

### PR DESCRIPTION
**Pull request by Siupahc**
Added navigation as we did in the bootcamp, in NavigationActions.

Added a BottomNavigationMenu composable, which use material3 NavigationBarItem instead of material BottomNavigationItem (basically the same)

Addded a MainScreen, as well as an Achievements and Video Screen, all basically empty except for the navigation bar

Also changed some versions in libs.versions.toml, such as the kotlin version used (1.8.10 -> 1.9) or the target JVM (1.8 ->11).
Thoses are the same versions that we had in the bootcamp, and without those we can't use some testing tools.

In MainActivity I setup the navigation, and created a placeholder for the login screen.

Basically, it's the same navigation system as in the bootcamp

### CODE REVIEW

**Comment by Siupahc** [comment]
Code coverage is quite bad mainly because MainActivity isn't tested (doesn't make sense to test it rn) and there's so few code, that some useless copy pasted code from the bootcamp (see in NavigationActions, in navigateTo) take a significant part and can't be tested. I kept this code because, well the teacher probably put it for a good reason.

**Review by Shayan105 — CHANGES_REQUESTED** [review]
Need to modularize the botomNavigationMenu instantiation for a clearer code

File: app/src/main/java/com/android/sample/ui/navigation/BottomNavigationMenu.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+
+@Composable
```
Comment (Siupahc) [inline app/src/main/java/com/android/sample/ui/navigation/BottomNavigationMenu.kt:13]: Maybe add a composable representing the standard navigation bar that we will use the most, and make use of default parameter to avoid boilerplate code. Would need to pass navigationAction as param though

**Review by Shayan105 — APPROVED** [review]
Code is quite the same as the bootcamp. And tests are good. Approved

File: app/src/main/java/com/android/sample/ui/video/VideoScreen.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.android.sample.ui.navigation.LIST_OF_TOP_LEVEL_DESTINATIONS
+import com.android.sample.ui.navigation.NavigationActions
+
+@Composable
+fun VideoScreen(navigationActions: NavigationActions) {
+  Scaffold(
+      modifier = Modifier.testTag("videoScreen"),
+      content = { padding -> Text("Video Screen", modifier = Modifier.padding(padding)) },
+      bottomBar = {
+        BottomNavigationMenu(
```
[the remark this answers is not in the document]
Comment (Siupahc) [reply app/src/main/java/com/android/sample/ui/video/VideoScreen.kt:19]: Yes we could. We could create a composable representing the main navigation bar that we will mostly use, and use default parameters to give him LIST_OF_TOP_LEVEL_DESTINATIONS, and correct behaviour for onTabSelected, but this would require to pass NavigationActions as params. Don't know if it's worth it to save 2 lines. Perhaps, good idea.
